Library/Homebrew/livecheck/strategy.rb

```diff
@@ -119,25 +119,48 @@ def self.page_headers(url)
 
       # Fetches the content at the URL and returns a hash containing the
       # content and, if there are any redirections, the final URL.
+      # If `curl` encounters an error, the hash will contain a `:messages`
+      # array with the error message instead.
       #
       # @param url [String] the URL of the content to check
+      # @param user_agent [Symbol, String] the symbol for a corresponding user
+      #   agent or a user agent string
       # @return [Hash]
-      sig { params(url: String).returns(T::Hash[Symbol, T.untyped]) }
-      def self.page_content(url)
+      sig { params(url: String, user_agent: T.any(Symbol, String)).returns(T::Hash[Symbol, T.untyped]) }
+      def self.page_content(url, user_agent: :default)
```

Comment:
When is this `user_agent` parameter used?

Reply:
This is intended to allow configuration of the user agent in a `livecheck` block in the future but that requires changes to the DSL (and livecheck) that won't happen here. This `user_agent` parameter isn't doing anything in this PR other than laying the groundwork for this.

I can remove it from this PR for the sake of simplicity and add it later when I create a PR for the DSL change. The `curl` user agent is set to `:default` when `user_agent` isn't provided to `curl_args`, so it would act the same without this in the interim time.